bark/world/world.cpp

```diff
@@ -61,17 +61,20 @@ void World::PlanAgents(const float& delta_time) {
   WorldPtr current_world(this->Clone());
   const float inc_world_time = world_time_ + delta_time;
   for (auto agent : agents_) {
-    ObservedWorld observed_world(current_world, agent.first);
-    agent.second->PlanBehavior(delta_time, observed_world);
-    if (agent.second->GetBehaviorStatus() == BehaviorStatus::VALID)
-      agent.second->PlanExecution(inc_world_time);
+    if (agent.second->IsValidAtTime(world_time_)) {
+      ObservedWorld observed_world(current_world, agent.first);
+      agent.second->PlanBehavior(delta_time, observed_world);
+      if (agent.second->GetBehaviorStatus() == BehaviorStatus::VALID)
+        agent.second->PlanExecution(inc_world_time);
+    }
   }
 }
 
 void World::Execute(const float& world_time) {
   using models::dynamic::StateDefinition::TIME_POSITION;
   for (auto agent : agents_) {
-    if (agent.second->GetBehaviorStatus() == BehaviorStatus::VALID &&
+    if (agent.second->IsValidAtTime(world_time_) &&
```

Comment:
not required as the behavior model would set invalid right?

Reply:
I needed a separate mechanism to let agents start later that, as there might be a trajectory from the dataset, but it might not be within the map yet. if you want to start planning from there, that's a problem. 
Essentially, lifting the mechanism of checking validity to the agent level is much more powerful / convenient. We might consider removing the BehaviorStatus.

Reply:
yeah, maybe we should discuss this. From my perspective I would not have a problem with dropping the behavior status then